### PR TITLE
Xplat sdk

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: BuildAndTest.proj [hunk 1]
```diff
@@ -2,13 +2,13 @@
 <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
 <Project ToolsVersion="14.0" DefaultTargets="BuildAndTest" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  <Import Project="$(MSBuildThisFileDirectory)build\Targets\Versions.props" />
+  <Import Project="$(MSBuildThisFileDirectory)build/Targets/Versions.props" />
```
Comment (jaredpar) [inline BuildAndTest.proj]: Is this change necessary?  If not I'd prefer to stay `\` as the majority of our files have that approach.

File: Makefile [hunk 2]
```diff
@@ -2,91 +2,67 @@ SHELL = /usr/bin/env bash
...
-DOTNET_VERSION = 1.0.0-preview3-003223
-NUGET_VERSION = 3.5.0-beta2
-NUGET_EXE = $(shell pwd)/nuget.exe
+DOTNET_VERSION = 1.0.1
+DOTNET = $(BINARIES_PATH)/dotnet-cli/dotnet
+TARGET_FX = netcoreapp1.1
 
 MSBUILD_ADDITIONALARGS := /v:m /fl /fileloggerparameters:Verbosity=normal /p:Configuration=$(BUILD_CONFIGURATION)
 
+
```
Comment (jaredpar) [inline Makefile]: Double blank line.

**Comment by jaredpar** [comment]
CC @ellismg to see how we're approaching building on Unix using the SDK

File: Makefile [hunk 3]
```diff
@@ -2,91 +2,67 @@ SHELL = /usr/bin/env bash
...
 ifeq ($(OS_NAME),Linux)
-	MSBUILD_ADDITIONALARGS := $(MSBUILD_ADDITIONALARGS) /p:BaseNuGetRuntimeIdentifier=$(shell . /etc/os-release && echo $$ID.$$VERSION_ID)
+	RUNTIME_ID  := $(shell . /etc/os-release && echo $$ID.$$VERSION_ID)-x64
```
Comment (jaredpar) [inline Makefile]: Why are there two spaces between `RUNTIME_ID` and `:=`?  Can't tell if this is necessary or accident.  Make file spacing is important but don't think for this case.  This pattern is repeated a couple times in this file.

File: build/Targets/Settings.props [hunk 4]
```diff
@@ -224,16 +230,6 @@
...
 
-  <!--
-    Roslyn uses the WriteCodeFragment task to output generated assembly attributes.  In MSBuild 14.0 update 1 this task
-    produces non-deterministic output.  This is fixed in update 2 but until that is released we use a hand patched
-    version of the Task that is deterministic.
-
-    MSBuild bug: https://github.com/Microsoft/msbuild/issues/408
-    Bug tracking removing this: https://github.com/dotnet/roslyn/issues/8421
-  -->
-  <UsingTask TaskName="Roslyn.MSBuild.Util.WriteCodeFragmentEx" AssemblyFile="$(RoslynBuildUtilFilePath)" />
```
Comment (jaredpar) [inline build/Targets/Settings.props]: Should be able to delete the NuGet package backing this task entirely now.  Didn't see the delete in Packages.props.

File: build/Targets/Imports.targets [hunk 5]
```diff
@@ -62,8 +62,8 @@
       <PropertyGroup>
         <AllowedReferenceRelatedFileExtensions>$(AllowedReferenceRelatedFileExtensions);.pdb</AllowedReferenceRelatedFileExtensions>
         <CopyNuGetImplementations>true</CopyNuGetImplementations>
-        <OutputPath Condition="'$(TargetFrameworkIdentifier)' == '.NETPortable'">$(OutputPath)CoreClrTest</OutputPath>
-        <OutputPath Condition="'$(TargetFrameworkIdentifier)' != '.NETPortable'">$(OutputPath)UnitTests\Portable\</OutputPath>
+        <OutputPath Condition="$(TargetFramework.StartsWith('netcoreapp'))">$(OutputPath)CoreClrTest</OutputPath>
+        <OutputPath Condition="!$(TargetFramework.StartsWith('netcoreapp'))">$(OutputPath)UnitTests\Portable\</OutputPath>
```
Comment (jaredpar) [inline build/Targets/Imports.targets]: FYI: fixed this in the PR I have out.

File: build/Targets/Settings.props [hunk 6]
```diff
@@ -74,11 +73,18 @@
...
+    <RoslynNetSdkRootPath>$(MSBuildSDKsPath)\Microsoft.NET.Sdk\Sdk\</RoslynNetSdkRootPath> 
     <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
   </PropertyGroup>
 
+  <ItemGroup Label="ToolsetPackages">
+    <PackageReference Include="$(ToolsetCompilerPackageName)">
+      <Version>$(ToolsetCompilerPackageVersion)</Version>
+      <ExcludeAssets>all</ExcludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (jaredpar) [inline build/Targets/Settings.props]: Why do we need this?  It's adding an implicit PackageReference to every item that build in our repo.  That doesn't feel right.

File: build/Targets/Settings.props [hunk 7]
```diff
@@ -13,8 +13,10 @@
     <NuGetToolPath>$(RepoRoot)nuget.exe</NuGetToolPath>
     <ToolsetPackagesDir>$(RepoRoot)build\ToolsetPackages\</ToolsetPackagesDir>
     <ToolsetPackagesSemaphore>$(ToolsetPackagesDir)toolsetpackages.semaphore</ToolsetPackagesSemaphore>
-    <ToolsetCompilerPackageName>Microsoft.Net.Compilers</ToolsetCompilerPackageName>
-    <ToolsetCompilerPackageVersion>2.0.1</ToolsetCompilerPackageVersion>
+    <ToolsetCompilerPackageName Condition="'$(OS)' == 'Windows_NT'">Microsoft.Net.Compilers</ToolsetCompilerPackageName>
+    <ToolsetCompilerPackageName Condition="'$(OS)' != 'Windows_NT'">Microsoft.NETCore.Compilers</ToolsetCompilerPackageName>
+    <ToolsetCompilerPackageVersion Condition="'$(OS)' == 'Windows_NT'">2.0.1</ToolsetCompilerPackageVersion>
+    <ToolsetCompilerPackageVersion Condition="'$(OS)' != 'Windows_NT'">2.3.0-dev-56735-00</ToolsetCompilerPackageVersion>
```
Comment (jaredpar) [inline build/Targets/Settings.props]: Doesn't look like these properties are used here.  Can we group them together with the `$(ToolsetCmpilerPropsFilePath)` property on line 40?

**Comment by agocke** [comment]
@jaredpar I'll look through your comments now -- I was fixing up the build. This is now expected to pass all tests.

File: build/Targets/Versions.props [hunk 8]
```diff
@@ -49,6 +49,8 @@
     <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' == '' AND '$(OS)' == 'Windows_NT'">$(UserProfile)/.nuget/packages/</NuGetPackageRoot>
     <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' == '' AND '$(OS)' != 'Windows_NT'">$(HOME)/.nuget/packages/</NuGetPackageRoot>
     <NuGetPackagesDirectory>$(NuGetPackageRoot)</NuGetPackagesDirectory>
+
+    <Version Condition="$(Version) == ''">$(NuGetPerBuildPreReleaseVersion)</Version>
```
Comment (jaredpar) [inline build/Targets/Versions.props]: This is a dangerous property to set.  NuGet uses it for a lot of different scenarios.  Why do we need it here?

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj [hunk 9]
```diff
@@ -5,20 +5,15 @@
...
-    <RuntimeIdentifier>x64</RuntimeIdentifier>
-    <ProjectGuid>{59BABFC3-C19B-4472-A93D-3DD3835BC219}</ProjectGuid>
     <OutputType>Exe</OutputType>
     <AssemblyName>DeployCoreClrTestRuntime_DoNotUse</AssemblyName>
     <Prefer32Bit>false</Prefer32Bit>
     <LargeAddressAware>true</LargeAddressAware>
-    <AutoGenerateBindingRedirects>True</AutoGenerateBindingRedirects>
-    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <TargetFramework>netcoreapp1.1</TargetFramework>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (jaredpar) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: Why did we need this change?

File: BuildAndTest.proj [hunk 10]
```diff
@@ -17,14 +17,16 @@
...
+    <OutputDirectory>$(MSBuildThisFileDirectory)Binaries\$(Configuration)\</OutputDirectory>
     <RunTestArgs>$(RunTestArgs) -log:&quot;$(OutputDirectory)\runtests.log&quot;</RunTestArgs>
     <CoreClrTestDirectory>$(OutputDirectory)\CoreClrTest</CoreClrTestDirectory>
     <MSBuildCommonProperties>
       RestorePackages=false;
       TreatWarningsAsErrors=true;
       DeployExtension=false;
     </MSBuildCommonProperties>
+    <TestDesktop Condition="'$(TestDesktop)' == ''">true</TestDesktop>
+    <TestCoreClr Condition="'$(TestCoreClr)' == ''">true</TestCoreClr>
```
Comment (agocke) [inline BuildAndTest.proj]: Didn't you just do an in-person survey of our team and found that most people who ran tests locally did `msbuild BuildAndTest.proj`? 😛

File: BuildAndTest.proj [hunk 11]
```diff
@@ -79,28 +81,41 @@
           SkipUnchangedFiles="true"
           UseHardlinksIfPossible="true" />
 
-    <MakeDir Directories="$(CoreClrTestDirectory)\xUnitResults" />
+    <MakeDir Directories="$(CoreClrTestDirectory)/xUnitResults" />
+
+    <!-- Publish binaries for CoreClr test -->
+    <MSBuild BuildInParallel="true"
+             Projects="src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj"
+             Properties="RuntimeIdentifier=win7-x64;PublishDir=$(CoreClrTestDirectory)"
```
Comment (agocke) [inline BuildAndTest.proj]: The xplat builds currently use the makefile for testing, so this is never run outside of Windows. It may be worthwhile to consider unifying these soon.

File: build/Targets/Versions.props (hunk 8 above)
Comment (agocke) [inline build/Targets/Versions.props]: Downgrade warnings otherwise since referencing NETCore.App includes Roslyn. The projects are seen as 1.0.0 as default and the NETCore.App version is >= 1.3.0, so 1.0.0 is always a downgrade.

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj (hunk 9 above)
Comment (agocke) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: NuGet 4.0 seems to require it to properly unify the package graph.

File: BuildAndTest.proj (hunk 10 above)
Comment (jaredpar) [inline BuildAndTest.proj]: Yep.  We can teach them to use `Test.cmd`.  I actually want to delete this file entirely because it's just a glorified script at this point.

File: build/Targets/Versions.props (hunk 8 above)
Comment (jaredpar) [inline build/Targets/Versions.props]: We shouldn't hide downgrade warnings this way.  In the cases where that warning happens we can just include a different package.  Look at the assets file, see which pcakage brings in `dotnet.exe`.  Reference that instead.  

@nguerrera assured me this will work.

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj (hunk 9 above)
Comment (jaredpar) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: Did we dig into that?  Plenty of our other graphs unified under 1.1.

File: src/Compilers/Core/Portable/CodeAnalysis.csproj [hunk 12]
```diff
@@ -20,11 +20,11 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
   <ItemGroup>
-    <Content Include="$(NuGetPackageRoot)\Microsoft.DiaSymReader.Native\$(MicrosoftDiaSymReaderNativeVersion)\runtimes\win\native\Microsoft.DiaSymReader.Native.x86.dll">
+    <Content Include="$(NuGetPackageRoot)\microsoft.diasymreader.native\$(MicrosoftDiaSymReaderNativeVersion)\runtimes\win\native\Microsoft.DiaSymReader.Native.x86.dll">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
       <Visible>false</Visible>
     </Content>
-    <Content Include="$(NuGetPackageRoot)\Microsoft.DiaSymReader.Native\$(MicrosoftDiaSymReaderNativeVersion)\runtimes\win\native\Microsoft.DiaSymReader.Native.amd64.dll">
+    <Content Include="$(NuGetPackageRoot)\microsoft.diasymreader.native\$(MicrosoftDiaSymReaderNativeVersion)\runtimes\win\native\Microsoft.DiaSymReader.Native.amd64.dll">
```
Comment (tmat) [inline src/Compilers/Core/Portable/CodeAnalysis.csproj]: >microsoft.diasymreader.native [](start = 42, length = 29)

Does nuget now always lowercase the name?

File: src/Compilers/CSharp/CscCore/CscCore.csproj [hunk 13]
```diff
@@ -30,16 +29,11 @@
       <Project>{B501A547-C911-4A05-AC6E-274A50DFF30E}</Project>
       <Name>CSharpCodeAnalysis</Name>
     </ProjectReference>
-    <PackageReference Include="System.AppContext" Version="$(SystemAppContextVersion)" />
-    <PackageReference Include="System.Console" Version="$(SystemConsoleVersion)" />
-    <PackageReference Include="System.Diagnostics.Process" Version="$(SystemDiagnosticsProcessVersion)" />
+  </ItemGroup>
+  <ItemGroup Label="Package References">
```
Comment (tmat) [inline src/Compilers/CSharp/CscCore/CscCore.csproj]: >Label="Package References" [](start = 13, length = 26)

Labels are redundant.

File: src/Compilers/Core/Portable/CodeAnalysis.csproj (hunk 12 above)
Comment (agocke) [inline src/Compilers/Core/Portable/CodeAnalysis.csproj]: Yes

File: src/Compilers/Core/Portable/CodeAnalysis.csproj [hunk 14]
```diff
@@ -816,6 +808,7 @@
     <Compile Include="Xml\XmlCharType.cs" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.DiaSymReader.Native" Version="$(MicrosoftDiaSymReaderNativeVersion)" />
```
Comment (tmat) [inline src/Compilers/Core/Portable/CodeAnalysis.csproj]: Are you sure this works? Does it deploy both x86 and amd64 versions?

Comment (tmat) [inline src/Compilers/Core/Portable/CodeAnalysis.csproj]: Specifically, when building Emit unit tests.

File: build/Targets/Versions.props (hunk 8 above)
Comment (agocke) [inline build/Targets/Versions.props]: But we also need to deploy for the shared framework. @nguerrera is there any way to generate a proper `runtimeconfig.json` and `deps.json` without referencing `Microsoft.NETCore.App`?

Comment (nguerrera) [inline build/Targets/Versions.props]: The downgrade warnings are bug NuGet/Home#4620.

> @nguerrera is there any way to generate a proper runtimeconfig.json and deps.json without referencing Microsoft.NETCore.App?

I just did a quick test I am able to do this without referencing Microsoft.NETCore.App. Can you file a bug with repro steps on dotnet/sdk?

Comment (nguerrera) [inline build/Targets/Versions.props]: Ah, I get a repro using netcoreapp1.1 and v1.0 SDK. runtimeconfig.json comes out empty. Please do file the bug.

Comment (nguerrera) [inline build/Targets/Versions.props]: Ah, I get a repro using netcoreapp1.1 and v1.0 SDK. runtimeconfig.json comes out empty. Please do file the bug.

Comment (nguerrera) [inline build/Targets/Versions.props]: (My quick test was faulty, this does not work with 2.0 either) :(

File: src/Compilers/Core/Portable/CodeAnalysis.csproj (hunk 14 above)
Comment (agocke) [inline src/Compilers/Core/Portable/CodeAnalysis.csproj]: Seems to work for unit tests, but not for VSIX. Not sure why. I'll try something else.

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj (hunk 9 above)
Comment (agocke) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: OK, it doesn't fail, it just spews a lot of info from the task that does the unification. That's by design.

However, why would we reference NETCore.App 2.0 package and have a netcoreapp 1.1 TFM?

File: build/Targets/Versions.props (hunk 8 above)
Comment (agocke) [inline build/Targets/Versions.props]: https://github.com/dotnet/sdk/issues/1159

File: BuildAndTest.proj [hunk 15]
```diff
@@ -99,8 +105,15 @@
       <CoreRunArgs>$(CoreClrTestDirectory)\xunit.console.netcore.exe @(CoreTestAssemblies, ' ') -parallel all -xml $(CoreClrTestDirectory)\xUnitResults\TestResults.xml</CoreRunArgs>
       <RunTestsExe>Binaries\$(Configuration)\Exes\RunTests\RunTests.exe</RunTestsExe>
       <RunTestsArgs>$(NuGetPackageRoot)\xunit.runner.console\$(xunitrunnerconsoleVersion)\tools $(RunTestArgs) @(TestAssemblies, ' ')</RunTestsArgs>
+      <VsixExpInstallerExe>$(NuGetPackageRoot)\roslyntools.microsoft.vsixexpinstaller\$(RoslynToolsMicrosoftVSIXExpInstallerVersion)\tools\VsixExpInstaller.exe</VsixExpInstallerExe>
+      <VsixExpInstallerArgs>-rootSuffix:RoslynDev -vsInstallDir:"$([System.IO.Path]::GetFullPath('$(MSBuildBinPath)\..\..\..'))"</VsixExpInstallerArgs>
     </PropertyGroup>
 
+    <!-- Publish binaries for CoreClr test -->
+    <MSBuild Projects="src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj"
```
Comment (jaredpar) [inline BuildAndTest.proj]: Still using `/` here.

File: Makefile [hunk 16]
```diff
@@ -2,91 +2,66 @@ SHELL = /usr/bin/env bash
...
-ifneq ($(SKIP_CROSSGEN),true)
-	build/scripts/crossgen.sh $(BOOTSTRAP_PATH)
-endif
+bootstrap: restore
+	export PATH="$(BINARIES_PATH)/dotnet-cli:$(PATH)" ; \
+	$(BUILD_CMD) src/Compilers/CSharp/CscCore && \
+	$(BUILD_CMD) src/Compilers/VisualBasic/VbcCore && \
+	mkdir -p $(BOOTSTRAP_PATH)/csc && mkdir -p $(BOOTSTRAP_PATH)/vbc && \
+	dotnet publish -r $(RUNTIME_ID) src/Compilers/CSharp/CscCore -o $(BOOTSTRAP_PATH)/csc && \
+	dotnet publish -r $(RUNTIME_ID) src/Compilers/VisualBasic/VbcCore -o $(BOOTSTRAP_PATH)/vbc
```
Comment (jaredpar) [inline Makefile]: There is an errant blank line here.  Is that intentional?  Not crisp on makefile rules but this seems wrong at a glance.

File: build/Targets/Packages.props [hunk 17]
```diff
@@ -3,220 +3,222 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (jaredpar) [inline build/Targets/Packages.props]: Why did this file change so much?  No way to review it.

Did you edit this on unix and as such change line endings?  If so can you change ti back so we can see the diff?

File: build/Targets/Versions.props (hunk 8 above)
Comment (jaredpar) [inline build/Targets/Versions.props]: Not sure what the conclusion is here.  Can we do this without MS.NetCore.app or not? 

If it is required we should move the `<Version>` element to the affected project files.  Setting it here unintentionally gives NuGet data to incorrectly process for every project in the system.  Should keep a work around like this as local as possible.

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj (hunk 9 above)
Comment (jaredpar) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: Still very unclear about this.  We use netcoreapp1.1 in literally every other portable EXE.  What is so special about this?  Seems like a bug.

File: build/Targets/Packages.props (hunk 17 above)
Comment (agocke) [inline build/Targets/Packages.props]: There was a bunch of strange whitespace. I think your tool did something busted. If you add ?w=1 it'll show the diff without whitespace changes.

**Comment by agocke** [comment]
@jaredpar All the other portable apps reference netcoreapp1.1 because they don't reference netstandard 2.0. Our tests do, so they need a netcoreapp 2.0 reference.

File: Makefile (hunk 16 above)
Comment (agocke) [inline Makefile]: Doesn't matter to Make, but I'll remove it anyway.

File: build/Targets/Packages.props (hunk 17 above)
Comment (jaredpar) [inline build/Targets/Packages.props]: Ah okay.  Looked at teh file and two pieces of feedback:

- Added a blank line, please remove 
- Roslyn.BuildUtil is still referenced and should be removed per earlier comment

**Comment by jaredpar** [comment]
> All the other portable apps reference netcoreapp1.1 because they don't reference netstandard 2.0. Our tests do, so they need a netcoreapp 2.0 reference.

I thought that the netstandard 2.0 binaries were used inside the tests, not as a referece that the test projects themselves depended on.

**Comment by agocke** [comment]
Both, but either way the Deploy project needs to reference to spit out the runtime assemblies for the tests to load.

File: BuildAndTest.proj [hunk 18]
```diff
@@ -79,17 +79,23 @@
           SkipUnchangedFiles="true"
           UseHardlinksIfPossible="true" />
 
-    <MakeDir Directories="$(CoreClrTestDirectory)\xUnitResults" />
+    <MakeDir Directories="$(CoreClrTestDirectory)/xUnitResults" />
+
+    <!-- Publish binaries for CoreClr test -->
+    <MSBuild BuildInParallel="true"
+             Projects="src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj"
```
Comment (jaredpar) [inline BuildAndTest.proj]: More forward slashes.

**Comment by jaredpar** [comment]
> Both, but either way the Deploy project needs the reference to spit out the runtime assemblies for the tests to load.

Ah yes.  The deployment problem 😦  That makes sense.

Still a bit confused about Both though as an answer.  Looking around I don't see a netstandard 2.0 `TargetFramework` entry in any of the project files.  We can sync up about that tomorrow.  Probably just me being tired and missing it.

File: BuildAndTest.proj (hunk 18 above)
Comment (agocke) [inline BuildAndTest.proj]: Woah, actually that's a bad merge. I'll delete.

Comment (jaredpar) [inline BuildAndTest.proj]: Score a victory for pedantic reviews around slash directions ;)

**Comment by agocke** [comment]
TestUtilities.CoreClr -- NETStandard.Library fixed version:
```
    <NETStandardLibraryFixedVersion>2.0.0-beta-25123-01</NETStandardLibraryFixedVersion>
```

Higher TFM isn't needed since we don't need any more APIs and TFM for libraries just controls what API surface area you're exposed to, not what you run on.

**Comment by jaredpar** [comment]
> TestUtilities.CoreClr -- NETStandard.Library fixed version:

Ah.  It's not in the actual test projects (wherce I was looking) but in the CoreCLR runtime driver.  Got it.

File: BuildAndTest.proj (hunk 18 above)
Comment (agocke) [inline BuildAndTest.proj]: I'm happy with the pedantry although I still strongly believe the slashes should go the other way.

File: src/Compilers/CSharp/Portable/CSharpCodeAnalysis.csproj [hunk 19]
```diff
@@ -14,6 +14,12 @@
     <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
     <ServiceablePackage>true</ServiceablePackage>
     <CodeAnalysisRuleSet>..\CSharpCodeAnalysisRules.ruleset</CodeAnalysisRuleSet>
+    <!-- This is required to prevent downgrade references from CscCore/VbcCode since we must
+         reference Microsoft.NETCore.App to target the shared framework (bug
+         https://github.com/dotnet/sdk/issues/1159) and NETCore.App
+         contains a reference to C# + VB CodeAnalysis. This ensures that the project
+         reference is a higher version than the NuGet reference. -->
+    <Version>$(NuGetPerBuildPreReleaseVersion)</Version>
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/CSharpCodeAnalysis.csproj]: See @dsplaisted comment here about using `<Version>` vs. `<PackageVersion>`

https://github.com/NuGet/Home/issues/4620#issuecomment-297907956

I'll let you evaluate if that concern is relevant for us here.

Comment (agocke) [inline src/Compilers/CSharp/Portable/CSharpCodeAnalysis.csproj]: We're passing an invalid assembly version as `Version` here, so I think it should be OK.

Comment (nguerrera) [inline src/Compilers/CSharp/Portable/CSharpCodeAnalysis.csproj]: If you're using the SDK's assembly info generation (GenerateAssemblyInfo is not false) and you haven't set an explicit AssemblyVersion, it we will be taken to be major.minor.patch.0 taken from semantic Version=major.minor.patch.

File: BuildAndTest.proj (hunk 18 above)
Comment (jaredpar) [inline BuildAndTest.proj]: The last commit said removing code from a bad merge yet I still see the `/` here.  Can we grep this file, ensure there are no `/` remaining and then push?  Prefer not to have another iteration on this file where we remove `/`.

File: src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj [hunk 20]
```diff
@@ -5,20 +5,14 @@
...
     <RuntimeIdentifiers>win7-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;osx.10.12-x64</RuntimeIdentifiers>
     <PackageTargetFallback>portable-net452;dotnet;netstandard1.6</PackageTargetFallback>
-    <NoStdLib>true</NoStdLib>
-    <RoslynProjectType>DeploymentTest</RoslynProjectType>
-    <RoslynUnitTestDir>CoreClrTest</RoslynUnitTestDir>
+    <SelfContained>true</SelfContained>
```
Comment (jaredpar) [inline src/Test/DeployCoreClrTestRuntime/DeployCoreClrTestRuntime.csproj]: What is this attribute?

File: BuildAndTest.proj (hunk 18 above)
Comment (agocke) [inline BuildAndTest.proj]: I'm pretty sure `</` is a necessary token in this file. :) I'll look for all stragglers though.
